tests/extract-references.js

```diff
@@ -0,0 +1,278 @@
+</dl>`,
+    res: {
+      normative: [{ name: "CSS-ALIGN-3", url: "https://drafts.csswg.org/css-align/" }],
+      informative: [{ name: "CSS-MULTICOL-1", url: "https://drafts.csswg.org/css-multicol/" }]
+    }
+  },
+
+  {
+    title: "does not get confused by further lists",
+    html: `
+<h2>Normative references</h2>
+<p>No references.</p>
+<h2>A few terms</h2>
+<dl>
+   <dt>A term</dt>
+   <dd>but not a ref</dd>
+</dl>`,
+    res: null
+  },
+
+  {
+    title: "extracts references defined in ul lists",
+    html: `
+<h2>11 References</h2>
+<h3>11.1 Normative References</h3>
+<ul>
+  <li>
+    <p><a href="https://aomediacodec.github.io/av1-spec/av1-spec.pdf">AV1</a> <strong>AV1 Bitstream &amp; Decoding Process Specification, Version 1.0.0 with Errata 1</strong>, January 2019.</p>
+  </li>
+</ul>
+<h3>11.2 Informative References</h3>
+<ul>
+  <li>
+    <p><a href="https://tools.ietf.org/html/rfc3711">RFC3711</a> <strong>The Secure Real-time Transport Protocol (SRTP)</strong>, M. Baugher, D. McGrew, M. Naslund, E. Carrara, and K. Norrman, March 2004.</p>
+  </li>
+</ul>`,
+    res: {
+      normative: [{ name: "AV1", url: "https://aomediacodec.github.io/av1-spec/av1-spec.pdf" }],
+      informative: [{ name: "RFC3711", url: "https://tools.ietf.org/html/rfc3711" }]
+    }
+  },
+
+  {
+    title: "looks for references in the last candidate section",
+    html: `
+<h2>Named character references</h2>
+<dl>
+  <dt>A name</dt>
+  <dd>Not a ref</dd>
+</dl>
+<h2>References</h2>
+<dl>
+  <dt id="refsABNF">[ABNF]</dt>
+  <dd><cite><a href="https://www.rfc-editor.org/rfc/rfc5234">Augmented BNF for Syntax Specifications: ABNF</a></cite>, D. Crocker, P. Overell. IETF.</dd>
+  <dt id="refsAPNG">[APNG]</dt>
+  <dd>(Non-normative) <cite><a href="https://wiki.mozilla.org/APNG_Specification">APNG Specification</a></cite>. S. Parmenter, V. Vukicevic, A. Smith. Mozilla.</dd>
+</dl>`,
+    res: {
+      normative: [{ name: "ABNF", url: "https://www.rfc-editor.org/rfc/rfc5234" }],
+      informative: [{ name: "APNG", url: "https://wiki.mozilla.org/APNG_Specification" }]
+    }
+  },
+
+  {
+    title: "does not extract nested links to sections of a reference",
+    html: `
+<h2>Normative references</h2>
+<ul>
+  <li>
+    <a href="https://unicode.org/reports/tr35/">Unicode Locale Data Markup Language (LDML)</a>
+    <ul>
+      <li>
+        <a href="https://unicode.org/reports/tr35/#Unicode_Language_and_Locale_Identifiers">Part 1 Core, Section 3 Unicode Language and Locale Identifiers</a>
+      </li>
+    </ul>
+  </li>
+</ul>`,
+    res: {
+      normative: [{ name: "Unicode Locale Data Markup Language (LDML)", url: "https://unicode.org/reports/tr35/" }],
+      informative: []
+    }
+  },
+
+  {
+    title: "does not use a nested URL as URL of a reference",
```

Comment:
why is that a feature?

Reply:
I reformulated a bit. That's mostly due to ECMA-402, to avoid extracting URLs that don't have title and that are not fully what the reference is all about, see previous comment.